documentation/modules/cruise-control/con-rebalance-goals-proposals.adoc

```diff
@@ -0,0 +1,180 @@
+
+[id="soft-hard-goals"]
+[discrete]
+=== Soft goals and hard goals
+
+The following table explains the differences between soft and hard goals in cluster rebalance calculations:
+
+[cols="3*",options="header",stripes="none",separator=¦]
+|===
+
+¦Rebalance goal type
+¦Description
+¦Example
+
+¦Soft goals
+
+¦Also known as "best effort" goals, soft goals do _not_ need to be satisfied in rebalance proposals. 
+However, they are included in rebalance calculations. 
+A rebalance proposal that satisfies _all_ hard goals but violates one or more soft goals is valid.
+
+Rebalance goals are treated as soft goals, unless designated as hard goals
+
+¦Attempt to distribute a topic's replicas evenly across the cluster (topic replica distribution goal). 
+Ignore this goal if doing so will enable all hard goals to be met.
+
+¦Hard goals
+
+¦Hard goals _must_ be satisfied in rebalance proposals. 
+In other words, a rebalance proposal that does _not_ satisfy all the designated hard goals is rejected by Cruise Control and not sent to the user.
+
+Designating hard goals is optional.
+¦All brokers in the cluster must have fewer than 3 replicas (replica capacity goal).
+
+|===
+
+[discrete]
+=== Default and custom rebalance goals
+
+When interacting with Cruise Control to generate rebalance proposals, you can use the _default rebalance goals_, which are defined on deployment, or set ad hoc _custom rebalance goals_. 
+Both types of rebalance goal can be configured to include hard goals as well as soft goals.  
+
+*Default rebalance goals and cached proposal*
+
+The default rebalance goals are set in the `Kafka` custom resource. 
+You specify a list of goals in `spec.cruiseControl.config.default.goals`.
+
+Unless you set custom rebalance goals, the default rebalance goals are used by Cruise Control to return a _cached rebalance proposal_. 
+The cached proposal reflects the current state of the Kafka cluster and is updated every 15 minutes by default. 
+
+To review the list of default goals, see link:https://github.com/linkedin/cruise-control/wiki/Configurations#analyzer-configurations[Analyzer Configurations^] in the Cruise Control Wiki.
+
+For more information on setting the default rebalance goals, see xref:ref-cruise-control-configuration-{context}[]. 
+
+*Custom rebalance goals* 
+
+Custom rebalance goals are set in the `spec.goals` section of a `KafkaClusterRebalance` resource and override the default rebalance goals. 
+If no custom rebalance goals are set, the default rebalance goals are used.
+
+Use custom rebalance goals to generate a rebalance proposal that addresses a particular optimization scenario. 
+For example, you might want to optimize leader replica distribution across the Kafka cluster without considering other goals, such as disk or CPU usage. 
+So, you create a `KafkaClusterRebalance` resource containing a custom rebalance goal for leader replica distribution only.
```

Comment:
We do need to clearly document that if they specify a list of goals in the `KafkaRebalance` resource that does not include all the configured `hard.goals` they will get an error. In that case they need to set `skiphardgoalscheck: true` in the resource.

Reply:
In which section of the resource? Please can you give an example configuration.